<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Pathway Analysis using the ClusterProfiler package in R | Portfolio DataScience of Khadija Zbair</title>
  <meta name="description" content="2 Pathway Analysis using the ClusterProfiler package in R | Portfolio DataScience of Khadija Zbair" />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Pathway Analysis using the ClusterProfiler package in R | Portfolio DataScience of Khadija Zbair" />
  <meta property="og:type" content="book" />




  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Pathway Analysis using the ClusterProfiler package in R | Portfolio DataScience of Khadija Zbair" />




<meta name="author" content="Khadija Zbair" />


<meta name="date" content="2024-05-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />


<link rel="prev" href="future-outlook.html"/>
<link rel="next" href="guerrilla-analytics.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<script src="libs/core-js-2.5.3/shim.min.js"></script>
<script src="libs/react-18.2.0/react.min.js"></script>
<script src="libs/react-18.2.0/react-dom.min.js"></script>
<script src="libs/reactwidget-1.0.0/react-tools.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="libs/reactable-binding-0.4.4/reactable.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">

  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
  <li class="toc-logo"><h3 class=".paddingtitel">Portfolio of Khadija Zbair</h3></li>
<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html"><i class="fa fa-check"></i>Curriculum vitae</a>
<ul>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#profile"><i class="fa fa-check"></i>Profile</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#education"><i class="fa fa-check"></i>Education</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#relevant-courses-overview"><i class="fa fa-check"></i>Relevant courses overview</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#skills"><i class="fa fa-check"></i>Skills</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#job-experience"><i class="fa fa-check"></i>Job experience</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#hobbies"><i class="fa fa-check"></i>Hobbies</a></li>
<li class="chapter" data-level="" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#contact"><i class="fa fa-check"></i>Contact</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="future-outlook.html"><a href="future-outlook.html"><i class="fa fa-check"></i><b>1</b> Future outlook</a></li>
<li class="chapter" data-level="2" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><i class="fa fa-check"></i><b>2</b> Pathway Analysis using the ClusterProfiler package in R</a>
<ul>
<li class="chapter" data-level="2.1" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#datasets"><i class="fa fa-check"></i><b>2.1</b> Datasets</a></li>
<li class="chapter" data-level="2.2" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#my-workflow"><i class="fa fa-check"></i><b>2.2</b> My workflow</a></li>
<li class="chapter" data-level="2.3" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#project-aim"><i class="fa fa-check"></i><b>2.3</b> Project aim</a></li>
<li class="chapter" data-level="2.4" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#clusterprofiler-package"><i class="fa fa-check"></i><b>2.4</b> Clusterprofiler package</a></li>
<li class="chapter" data-level="2.5" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#getting-into-clusterprofiler"><i class="fa fa-check"></i><b>2.5</b> Getting into Clusterprofiler…</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#supported-analysis"><i class="fa fa-check"></i><b>2.5.1</b> Supported analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#ora-and-gsea-with-clusterprofiler"><i class="fa fa-check"></i><b>2.6</b> ORA and GSEA with clusterProfiler</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#installation-clusterprofiler"><i class="fa fa-check"></i><b>2.6.1</b> Installation Clusterprofiler</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#gsea-with-drosphilia-annotation-data"><i class="fa fa-check"></i><b>2.7</b> GSEA with drosphilia annotation data</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#gathering-data"><i class="fa fa-check"></i><b>2.7.1</b> Gathering data</a></li>
<li class="chapter" data-level="2.7.2" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#prepare-input-for-the-go-term-ontology-enrichment-analysis-gene-list"><i class="fa fa-check"></i><b>2.7.2</b> Prepare Input for the GO term ontology enrichment analysis: Gene list</a></li>
<li class="chapter" data-level="2.7.3" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#go-term-enrichment-ontology-analysis"><i class="fa fa-check"></i><b>2.7.3</b> GO term enrichment ontology analysis</a></li>
<li class="chapter" data-level="2.7.4" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#kegg-gene-set-enrichment-analysis-what-pathways-are-enriched"><i class="fa fa-check"></i><b>2.7.4</b> KEGG Gene Set Enrichment Analysis: What pathways are enriched?</a></li>
<li class="chapter" data-level="2.7.5" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#prepare-input"><i class="fa fa-check"></i><b>2.7.5</b> Prepare Input</a></li>
<li class="chapter" data-level="2.7.6" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#create-gsekegg-object-what-pathways-are-enriched"><i class="fa fa-check"></i><b>2.7.6</b> Create gseKEGG Object: What pathways are enriched?</a></li>
<li class="chapter" data-level="2.7.7" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#visualising-the-most-enriched-pathways"><i class="fa fa-check"></i><b>2.7.7</b> Visualising the most enriched pathways</a></li>
<li class="chapter" data-level="2.7.8" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#pathview"><i class="fa fa-check"></i><b>2.7.8</b> Pathview</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#ora-with-the-airway-dataset"><i class="fa fa-check"></i><b>2.8</b> ORA with the airway dataset</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#go-term-enrichment-ontology-analysis-1"><i class="fa fa-check"></i><b>2.8.1</b> GO term enrichment ontology analysis</a></li>
<li class="chapter" data-level="2.8.2" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#visualize-enriched-go-terms"><i class="fa fa-check"></i><b>2.8.2</b> Visualize enriched GO-terms</a></li>
<li class="chapter" data-level="2.8.3" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#kegg-pathway-enrichment-analyse"><i class="fa fa-check"></i><b>2.8.3</b> KEGG pathway enrichment analyse</a></li>
<li class="chapter" data-level="2.8.4" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#visualizing-the-enriched-pathways"><i class="fa fa-check"></i><b>2.8.4</b> Visualizing the enriched pathways</a></li>
<li class="chapter" data-level="2.8.5" data-path="pathway-analysis-using-the-clusterprofiler-package-in-r.html"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#pathview-1"><i class="fa fa-check"></i><b>2.8.5</b> Pathview</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="guerrilla-analytics.html"><a href="guerrilla-analytics.html"><i class="fa fa-check"></i><b>3</b> Guerrilla Analytics</a></li>
<li class="chapter" data-level="4" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html"><i class="fa fa-check"></i><b>4</b> C.elegans plate experiment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#importing-and-inspecting-the-data"><i class="fa fa-check"></i><b>4.1</b> Importing and inspecting the data</a></li>
<li class="chapter" data-level="4.2" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#checking-and-correcting-the-datatypes"><i class="fa fa-check"></i><b>4.2</b> Checking and correcting the datatypes</a></li>
<li class="chapter" data-level="4.3" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#scatterplot-with-log10-and-jitter"><i class="fa fa-check"></i><b>4.3</b> Scatterplot with log10 and jitter</a></li>
<li class="chapter" data-level="4.4" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#statistical-analysis"><i class="fa fa-check"></i><b>4.4</b> Statistical analysis</a></li>
<li class="chapter" data-level="4.5" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html#normalizing-the-controlnegative-data-and-plotting"><i class="fa fa-check"></i><b>4.5</b> Normalizing the ‘controlNegative’ data and plotting</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html"><i class="fa fa-check"></i><b>5</b> Reproducible Research: Peer review</a>
<ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#choosing-the-right-article"><i class="fa fa-check"></i><b>5.1</b> Choosing the right article</a></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#reference-to-the-article"><i class="fa fa-check"></i><b>5.2</b> Reference to the article</a></li>
<li class="chapter" data-level="5.3" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#transparicy-criteria"><i class="fa fa-check"></i><b>5.3</b> Transparicy criteria</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#general-aim"><i class="fa fa-check"></i><b>5.3.1</b> General aim</a></li>
<li class="chapter" data-level="5.3.2" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#methodology"><i class="fa fa-check"></i><b>5.3.2</b> Methodology</a></li>
<li class="chapter" data-level="5.3.3" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#results"><i class="fa fa-check"></i><b>5.3.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#conlusion"><i class="fa fa-check"></i><b>5.4</b> Conlusion</a></li>
<li class="chapter" data-level="5.5" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#reproducing-the-visuals-of-a-paper"><i class="fa fa-check"></i><b>5.5</b> Reproducing the visuals of a paper</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="reproducible-research-peer-review.html"><a href="reproducible-research-peer-review.html#the-code"><i class="fa fa-check"></i><b>5.5.1</b> The code</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-package.html"><a href="r-package.html"><i class="fa fa-check"></i><b>6</b> R-package</a>
<ul>
<li class="chapter" data-level="6.1" data-path="r-package.html"><a href="r-package.html#the-demo"><i class="fa fa-check"></i><b>6.1</b> The Demo</a></li>
<li class="chapter" data-level="6.2" data-path="r-package.html"><a href="r-package.html#r-markdown-driven-development"><i class="fa fa-check"></i><b>6.2</b> R-markdown driven development</a></li>
<li class="chapter" data-level="6.3" data-path="r-package.html"><a href="r-package.html#package-hanaahrtidy"><i class="fa fa-check"></i><b>6.3</b> Package “HanaahRTidy”</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="r-package.html"><a href="r-package.html#choosing-a-name"><i class="fa fa-check"></i><b>6.3.1</b> Choosing a name</a></li>
<li class="chapter" data-level="6.3.2" data-path="r-package.html"><a href="r-package.html#overview-package"><i class="fa fa-check"></i><b>6.3.2</b> Overview package</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-package.html"><a href="r-package.html#functionality"><i class="fa fa-check"></i><b>6.4</b> Functionality</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html"><i class="fa fa-check"></i><b>7</b> Antimicrobial resistance projecticum</a>
<ul>
<li class="chapter" data-level="7.1" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#introduction-1"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#about-antimicrobial-resistance"><i class="fa fa-check"></i><b>7.2</b> About antimicrobial resistance</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#r-shiny-app"><i class="fa fa-check"></i><b>7.2.1</b> R shiny app</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="antimicrobial-resistance-projecticum.html"><a href="antimicrobial-resistance-projecticum.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html"><i class="fa fa-check"></i><b>8</b> Parameterized COVID-19 report</a>
<ul>
<li class="chapter" data-level="8.1" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html#inspecting-the-data"><i class="fa fa-check"></i><b>8.1</b> Inspecting the data</a></li>
<li class="chapter" data-level="8.2" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html#visualising-the-data"><i class="fa fa-check"></i><b>8.2</b> Visualising the data</a></li>
<li class="chapter" data-level="8.3" data-path="parameterized-covid-19-report.html"><a href="parameterized-covid-19-report.html#parameters"><i class="fa fa-check"></i><b>8.3</b> Parameters</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio DataScience of Khadija Zbair</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pathway-analysis-using-the-clusterprofiler-package-in-r" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Pathway Analysis using the ClusterProfiler package in R<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#pathway-analysis-using-the-clusterprofiler-package-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>For my independent project, I am dedicating my time, a total of 4 days, to learning a new skill: conducting pathway analysis. Understanding the molecular mechanisms underlying diseases is paramount for developing effective treatments.The plan is to experiment with KEGG pathway analysis to become proficient in utilizing Clusterprofiler, the R-package through which pathway analysis will be conducted.</p>
<div id="datasets" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Datasets<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I will apply KEGG pathway analysis to Drosophila melanogaster data (annotation data) and <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0099625">the airway data from Himes et al. (2014)</a>.
In the study, researchers examined changes in gene expression in four different types of human airway muscle cells that were exposed to dexamethasone, a medication used for asthma.
The study used RNAseq data from 8 samples of these cells.
For each cell type, there were samples that were treated with the medication and untreated controls.
They used a method called DESeq2 to identify which genes were expressed differently between the treated and untreated cells.</p>
<hr />
</div>
<div id="my-workflow" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> My workflow<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#my-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Day
</th>
<th style="text-align:left;">
Activities
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Day 1
</td>
<td style="text-align:left;">
<ol>
<li>
<b>Dataset Collection:</b> Figure out how the Clusterprofiler works and what input (format) it requires. Gather gene expression data. I chose the Drosophila annotation data and the airway data.
</li>
<li>
<b>Data Preprocessing:</b> Import and preprocess the datasets in R if needed.
</li>
</ol>
</td>
</tr>
<tr>
<td style="text-align:left;">
Day 2
</td>
<td style="text-align:left;">
<ol>
<li>
<b>Load DE Results:</b> Load the differential expression analysis results into the DESeq2 object for further analysis. This step is necessary for both GSEA and ORA.
</li>
<li>
<b>Differential Gene Expression Analysis results:</b> Obtain the DESeq2 object of the datasets and obtain the log2 fold change values and the gene identifiers.
</li>
<li>
<b>Filtering Significant Genes:</b> Select genes with statistically significant differential expression.
</li>
<li>
<b>Create Gene Lists:</b> Create gene lists of up- and down-regulated genes, including their associated log2 fold change values. Use the <code>bitr()</code> function from ClusterProfiler to convert gene IDs if needed.
</li>
<li>
<b>Visual Inspection:</b> Create visualizations such as volcano plots and heatmaps for inspecting DEGs.
</li>
</ol>
</td>
</tr>
<tr>
<td style="text-align:left;">
Day 3
</td>
<td style="text-align:left;">
<ol>
<li>
<b>GO Term Enrichment Analysis:</b> Conduct GO term enrichment analysis on the list of up-regulated genes using <code>enrichGO()</code> to identify biological processes associated with these genes.
</li>
<li>
<b>Visualize GO Enrichment:</b> Visualize the GO enrichment results using a dotplot or another visualization option from ClusterProfiler.
</li>
<li>
<b>Interpretation:</b> Analyze and interpret the biological significance of the identified GO terms.
</li>
</ol>
</td>
</tr>
<tr>
<td style="text-align:left;">
Day 4
</td>
<td style="text-align:left;">
<ol>
<li>
<b>KEGG Pathway Analysis:</b> Perform KEGG pathway analysis using <code>enrichKEGG()</code> to map the identified DEGs to relevant biological pathways.
</li>
<li>
<b>Visualize Enriched Pathways:</b> Utilize interactive tools like <code>browseKEGG()</code> or <code>pathview()</code> to explore and visualize enriched KEGG pathways.
</li>
<li>
<b>Pathway Gene Enrichment Analysis:</b> Conduct pathway gene set enrichment analysis using <code>gseKEGG()</code>.
</li>
<li>
<b>Identify Key Pathways:</b> Select pathways significantly enriched with DEGs.
</li>
<li>
<b>Summarize Results:</b> Write a summary of the identified pathways and try to explain them. Visualize them with <code>pathview()</code> or other visualization plots.
</li>
</ol>
</td>
</tr>
</tbody>
</table>
<hr />
</div>
<div id="project-aim" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Project aim<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#project-aim" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The aim is to acquire a new skill: conducting pathway analysis.
For this, I need to master the Clusterprofiler package.
Learning <code>clusterProfiler</code> automates gene expression analysis with KEGG pathway enrichment, gene set enrichment, and over-representation analysis.
Mastering it helps interpret complex datasets, crucial for understanding diseases like cancer.</p>
</div>
<div id="clusterprofiler-package" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Clusterprofiler package<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#clusterprofiler-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>clusterProfiler</code> package helps to analyse and visualize gene expression data to <strong>identify and compare important biological pathways and functions</strong>.
It works with data from thousands of species and helps identify which biological pathways and functions are important in your data using databases like KEGG and Gene Ontology (GO).</p>
<p>With <code>clusterProfiler</code>, you can:</p>
<p>- <strong>Find Important Pathways</strong>: Identify which biological pathways are over-represented in your data.</p>
<p>- <strong>Compare Conditions</strong>: Easily compare data from different treatments or time points to see what changes.</p>
<p>- <strong>Visualize Results</strong>: Use simple tools to explore and visualize pathways and functions.</p>
<p>I will focus on identifying pathways and visualizing them.
One way to do that is by pathview(), see figure 2.1.</p>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="Images/pathview.png" alt="Visualize the selected KEGG pathway using pathview(). Gene expression values can be assigned to a gradient color scale." width="70%" />
<p class="caption">
Figure 2.1: Visualize the selected KEGG pathway using pathview(). Gene expression values can be assigned to a gradient color scale.
</p>
</div>
</div>
<div id="getting-into-clusterprofiler" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Getting into Clusterprofiler…<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#getting-into-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As preparation on how to use the Clusterprofiler package to make a pathway analysis and to get an idea on the options this package has and I can use, I have the next overview:</p>
<div class="figure"><span style="display:block;" id="fig:dus"></span>
<img src="Images/clusterProfiler_Overview_Wu_et_al_2021.png" alt="An overview of the clusterprofiler package and the associated analysis and visualization. Clusterprofiler supports Gene set enrichment analysis and over representation analysis." width="90%" />
<p class="caption">
Figure 2.2: An overview of the clusterprofiler package and the associated analysis and visualization. Clusterprofiler supports Gene set enrichment analysis and over representation analysis.
</p>
</div>
<div id="supported-analysis" class="section level3 hasAnchor" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Supported analysis<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#supported-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As can been seen in figure 2.2, Clusterprofiler supports the following analysis:</p>
<ol style="list-style-type: decimal">
<li><strong>Over-Representation Analysis</strong>: Over-representation analysis (ORA) identifies biological pathways or functions that are disproportionately represented among differentially expressed genes. It determines if certain pathways have more genes than expected by chance. Alongside ORA and GSEA, KEGG pathway enrichment analysis can also be performed to identify and visualize enriched pathways.</li>
<li><strong>Gene Set Enrichment Analysis</strong>: Gene Set Enrichment Analysis (GSEA) checks if specific groups of genes show significant differences in expression between two conditions, helping identify important biological changes.</li>
</ol>
<p>I am planning on doing an over-representation analysis with clusterProfiler for the airway study and Gene Set Enrichment Analysis for drosphilia annotated data.</p>
<hr />
</div>
</div>
<div id="ora-and-gsea-with-clusterprofiler" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> ORA and GSEA with clusterProfiler<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#ora-and-gsea-with-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In general, the steps are:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Load the differential expression results (DE results) in the Deseq2 object</strong>: We need the results from the differential expression analysis to generate gene lists for further functional analysis.
The results of these analysis is stored in a Deseq2 object.
For GSEA and ORA this step is the same.</p></li>
<li><p><strong>Create gene list up and downregulated</strong>: Create different lists of genes based on their differential expression with associated log2 fold change values.
To organize the genes into lists for specific analyses, such as up-regulated and down-regulated genes.</p></li>
</ol>
<p>Function from clusterprofiler: bitr().
It can convert gene IDs.</p>
<ol start="3" style="list-style-type: decimal">
<li><strong>Perform GO enrichment analysis on up-regulated genes</strong>: Perform Gene Ontology (GO) enrichment analysis on the list of up-regulated genes. This is to understand which biological processes are influenced by the differentially expressed genes.</li>
</ol>
<p>Function from clusterprofiler: enrichGO() for ORA and gseGO() for GSEA.</p>
<ol start="4" style="list-style-type: decimal">
<li><strong>Visualize GO enrichment results using a dotplot</strong>: Visualize the GO enrichment results using a dotplot and maybe another visualisation. To easily interpret and communicate the results of the GO enrichment analysis.</li>
</ol>
<p>There are many visualisation options within Clusterprofiler that can be seen in figure 2.2.
Some are specifically for GSEA or ORA, others can be used for both.</p>
<ol start="5" style="list-style-type: decimal">
<li><strong>Performing KEGG pathway Enrichment Analysis</strong>: Conduct KEGG pathway over-representation analysis to identify which KEGG pathways are significantly enriched in the list of differentially expressed genes.</li>
</ol>
<p>Function from clusterprofiler: enrichKEGG().</p>
<ol start="6" style="list-style-type: decimal">
<li><strong>Visualize Enriched Pathways</strong>: Utilize interactive tools to explore enriched KEGG pathways to gain a deeper understanding of the pathway enrichment results in an interactive manner.</li>
</ol>
<p>Function from clusterprofiler: browseKEGG().
An alternative to this is <a href="https://bioconductor.org/packages/release/bioc/html/pathview.html">pathview(), by Luo and Brouwer (2013)</a> from the pathview package, this does not belong to Clusterprofiler.</p>
<ol start="7" style="list-style-type: decimal">
<li><strong>Pathway gene enrichment analysis</strong>: Conducting a pathway gene set enrichment analysis.</li>
</ol>
<p>Function from clusterprofiler: gseKEGG().</p>
<hr />
<div id="installation-clusterprofiler" class="section level3 hasAnchor" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Installation Clusterprofiler<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#installation-clusterprofiler" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The installation of the Clusterprofiler package is described <a href="https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb1-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb1-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb1-4" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;clusterProfiler&quot;</span>)</span></code></pre></div>
<p>First, I load all the necessary libraries:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-1" tabindex="-1"></a><span class="co"># Loading the required libraries</span></span>
<span id="cb2-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb2-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-4" tabindex="-1"></a><span class="fu">library</span>(airway)</span>
<span id="cb2-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-5" tabindex="-1"></a><span class="fu">library</span>(org.Dm.eg.db)</span>
<span id="cb2-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-6" tabindex="-1"></a><span class="fu">library</span>(pathview)</span>
<span id="cb2-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-8" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb2-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-9" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb2-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb2-10" tabindex="-1"></a><span class="fu">library</span>(HanaahRTidy)</span></code></pre></div>
<hr />
</div>
</div>
<div id="gsea-with-drosphilia-annotation-data" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> GSEA with drosphilia annotation data<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#gsea-with-drosphilia-annotation-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="gathering-data" class="section level3 hasAnchor" number="2.7.1">
<h3><span class="header-section-number">2.7.1</span> Gathering data<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#gathering-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I’m analyzing Drosophila melanogaster data, so I install and load the annotation package “org.Dm.eg.db” <a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">here</a>.
There are also other available annotations.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb3-1" tabindex="-1"></a><span class="co"># Setting the organism (I chose fruit fly for example ) </span></span>
<span id="cb3-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb3-2" tabindex="-1"></a>organism <span class="ot">=</span> <span class="st">&quot;org.Dm.eg.db&quot;</span></span>
<span id="cb3-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb3-4" tabindex="-1"></a><span class="co">#BiocManager::install(organism, character.only = TRUE)</span></span>
<span id="cb3-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb3-5" tabindex="-1"></a><span class="fu">library</span>(organism, <span class="at">character.only =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Before diving into the analysis, we first need to prepare the input data.
In this step, we’re reading in data obtained from differential expression analysis using DESeq2 named “drosphila_example_de.csv”, which includes important information such as log2 fold change values.
This file contains the results of a differential gene expression analysis.
I found this data specifically for the Drosophila melanogaster <a href="https://github.com/nasqar/ClusterProfShinyGSEA/blob/master/www/exampleData/drosphila_example_de.csv">here</a>.</p>
</div>
<div id="prepare-input-for-the-go-term-ontology-enrichment-analysis-gene-list" class="section level3 hasAnchor" number="2.7.2">
<h3><span class="header-section-number">2.7.2</span> Prepare Input for the GO term ontology enrichment analysis: Gene list<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#prepare-input-for-the-go-term-ontology-enrichment-analysis-gene-list" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>I want the log2 fold change values, because these values tell if genes differentiate in gene expression.
I also save the gene identifiers in a vector with their log2 fold change value.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-1" tabindex="-1"></a><span class="co"># Using my own function to import the drosphilia deseq2 dataset </span></span>
<span id="cb4-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-2" tabindex="-1"></a>drosphilia_data <span class="ot">&lt;-</span> <span class="fu">import_data</span>(<span class="st">&quot;/Users/kzzba/Downloads/bookdown_khadija/drosphila_example_de.csv&quot;</span>)</span>
<span id="cb4-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-4" tabindex="-1"></a><span class="co"># I want the log2 fold change  </span></span>
<span id="cb4-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-5" tabindex="-1"></a>input_gene_list <span class="ot">&lt;-</span> drosphilia_data<span class="sc">$</span>log2FoldChange</span>
<span id="cb4-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-6" tabindex="-1"></a></span>
<span id="cb4-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-7" tabindex="-1"></a><span class="co"># Assigning the gene names in X column of the drosophila_data dataset to the elements of the original_gene_list vector(log2 fold) to identify each gene.log2fold value and gene name now in original_gene_list. </span></span>
<span id="cb4-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-8" tabindex="-1"></a><span class="fu">names</span>(input_gene_list) <span class="ot">&lt;-</span> drosphilia_data<span class="sc">$</span>X</span>
<span id="cb4-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-9" tabindex="-1"></a></span>
<span id="cb4-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-10" tabindex="-1"></a><span class="co"># Remove the missing values </span></span>
<span id="cb4-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-11" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(input_gene_list)</span>
<span id="cb4-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-13" tabindex="-1"></a><span class="co"># The list in decreasing order (a must for clusterProfiler)</span></span>
<span id="cb4-14"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb4-14" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> <span class="fu">sort</span>(gene_list, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Now I have created a gene list (in the object name “gene_list”, that contains the gene identifiers with their log2 fold change value).
I proceed with gene set enrichment analysis to discover biological insights from the dataset.
Using the gseGO function, I identify enriched gene ontology (GO) terms associated with the differentially expressed genes.</p>
</div>
<div id="go-term-enrichment-ontology-analysis" class="section level3 hasAnchor" number="2.7.3">
<h3><span class="header-section-number">2.7.3</span> GO term enrichment ontology analysis<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#go-term-enrichment-ontology-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now onto the GO term enrichment ontology analysis, which identifies which biological functions, processes, or cellular components are overrepresented in the gene list, providing insights into their biological significance.</p>
<p><strong>Input</strong>: the gene list, a sorted list of genes based on log2 foldchange values from the expression analysis.</p>
<p><strong>Output</strong>: The top results are displayed using head(gene_set_enrichment), providing an overview of enriched GO terms.</p>
<details>
<summary>
Parameters
</summary>
<ul>
<li>
<strong>ont = “ALL”</strong>: Specifies the Gene Ontology (GO) categories to include. “ALL” covers Biological Process (BP), Molecular Function (MF), and Cellular Component (CC).
</li>
<li>
<strong>keyType</strong>: Defines the type of gene identifiers used. Here, ENSEMBL IDs are used.
</li>
<li>
<strong>nPerm</strong>: Sets the number of permutations to 10,000 for robust results.
</li>
<li>
<strong>minGSSize and maxGSSize</strong>: Specifies the minimum and maximum number of genes allowed in the gene sets.
</li>
<li>
<strong>pvalueCutoff</strong>: Uses a cutoff value of 0.05 to determine which results are considered significant.
</li>
</ul>
</details>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-1" tabindex="-1"></a>gene_set_enrichment <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList=</span>gene_list, </span>
<span id="cb5-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-2" tabindex="-1"></a>             <span class="at">ont =</span><span class="st">&quot;ALL&quot;</span>, </span>
<span id="cb5-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-3" tabindex="-1"></a>             <span class="at">keyType =</span> <span class="st">&quot;ENSEMBL&quot;</span>, </span>
<span id="cb5-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-4" tabindex="-1"></a>             <span class="at">nPerm =</span> <span class="dv">10000</span>, </span>
<span id="cb5-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-5" tabindex="-1"></a>             <span class="at">minGSSize =</span> <span class="dv">1</span>, </span>
<span id="cb5-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-6" tabindex="-1"></a>             <span class="at">maxGSSize =</span> <span class="dv">1000</span>, </span>
<span id="cb5-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-7" tabindex="-1"></a>             <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb5-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-8" tabindex="-1"></a>             <span class="at">verbose =</span> <span class="cn">TRUE</span>, </span>
<span id="cb5-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-9" tabindex="-1"></a>             <span class="at">OrgDb =</span> organism, </span>
<span id="cb5-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-10" tabindex="-1"></a>             <span class="at">pAdjustMethod =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb5-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-11" tabindex="-1"></a></span>
<span id="cb5-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-12" tabindex="-1"></a></span>
<span id="cb5-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-13" tabindex="-1"></a><span class="co"># The results </span></span>
<span id="cb5-14"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb5-14" tabindex="-1"></a><span class="fu">head</span>(gene_set_enrichment)</span></code></pre></div>
<pre><code>##            ONTOLOGY         ID
## GO:0007186       BP GO:0007186
## GO:0004930       MF GO:0004930
## GO:0007188       BP GO:0007188
## GO:0007218       BP GO:0007218
## GO:0030055       CC GO:0030055
## GO:0005925       CC GO:0005925
##                                                                          Description
## GO:0007186                              G protein-coupled receptor signaling pathway
## GO:0004930                                       G protein-coupled receptor activity
## GO:0007188 adenylate cyclase-modulating G protein-coupled receptor signaling pathway
## GO:0007218                                            neuropeptide signaling pathway
## GO:0030055                                                   cell-substrate junction
## GO:0005925                                                            focal adhesion
##            setSize enrichmentScore       NES       pvalue     p.adjust   qvalue
## GO:0007186     227      -0.5558057 -2.115394 0.0001420253 0.0001420253 0.142238
## GO:0004930     113      -0.5809564 -2.021148 0.0001548947 0.0001548947 0.142238
## GO:0007188      63      -0.6926563 -2.201787 0.0001630523 0.0001630523 0.142238
## GO:0007218      55      -0.6056430 -1.875694 0.0001650165 0.0001650165 0.142238
## GO:0030055      38      -0.7017319 -2.018735 0.0001709110 0.0001709110 0.142238
## GO:0005925      25      -0.7818698 -2.082128 0.0001746725 0.0001746725 0.142238
##            rank                  leading_edge
## GO:0007186  885 tags=25%, list=6%, signal=24%
## GO:0004930 1016 tags=32%, list=7%, signal=30%
## GO:0007188 1344 tags=41%, list=9%, signal=38%
## GO:0007218  569 tags=35%, list=4%, signal=33%
## GO:0030055 1342 tags=32%, list=9%, signal=29%
## GO:0005925 1157 tags=36%, list=8%, signal=33%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            core_enrichment
## GO:0007186 FBgn0050361/FBgn0038880/FBgn0040506/FBgn0001104/FBgn0050340/FBgn0260446/FBgn0022710/FBgn0000253/FBgn0053639/FBgn0031275/FBgn0004842/FBgn0004573/FBgn0004622/FBgn0052547/FBgn0033932/FBgn0027794/FBgn0038140/FBgn0050106/FBgn0024944/FBgn0052683/FBgn0266137/FBgn0263131/FBgn0028956/FBgn0010223/FBgn0038653/FBgn0011582/FBgn0038542/FBgn0086704/FBgn0085380/FBgn0004784/FBgn0265959/FBgn0030437/FBgn0267252/FBgn0261953/FBgn0013767/FBgn0039396/FBgn0038980/FBgn0032048/FBgn0037976/FBgn0031302/FBgn0264002/FBgn0031770/FBgn0037408/FBgn0035092/FBgn0036260/FBgn0033744/FBgn0035385/FBgn0266429/FBgn0085413/FBgn0260753/FBgn0019985/FBgn0045038/FBgn0261549/FBgn0034636/FBgn0039419/FBgn0052843
## GO:0004930                                                                                                                                                                                                                                                 FBgn0263116/FBgn0035331/FBgn0050361/FBgn0038880/FBgn0050340/FBgn0260446/FBgn0053639/FBgn0031275/FBgn0004842/FBgn0004573/FBgn0004622/FBgn0052547/FBgn0033932/FBgn0038140/FBgn0050106/FBgn0024944/FBgn0266137/FBgn0028956/FBgn0038653/FBgn0011582/FBgn0038542/FBgn0025680/FBgn0030437/FBgn0039396/FBgn0038980/FBgn0264002/FBgn0031770/FBgn0037408/FBgn0036260/FBgn0033744/FBgn0035385/FBgn0266429/FBgn0260753/FBgn0019985/FBgn0039419/FBgn0052843
## GO:0007188                                                                                                                                                                                                                                                                                                                                                                         FBgn0040507/FBgn0050054/FBgn0000037/FBgn0037546/FBgn0263116/FBgn0035331/FBgn0040506/FBgn0001104/FBgn0260446/FBgn0022710/FBgn0004573/FBgn0033932/FBgn0266137/FBgn0263131/FBgn0010223/FBgn0038653/FBgn0011582/FBgn0038542/FBgn0030437/FBgn0038980/FBgn0032048/FBgn0264002/FBgn0033744/FBgn0260753/FBgn0039419/FBgn0052843
## GO:0007218                                                                                                                                                                                                                                                                                                                                                                                                                                                             FBgn0011581/FBgn0035331/FBgn0038880/FBgn0004842/FBgn0004622/FBgn0052547/FBgn0038140/FBgn0050106/FBgn0013767/FBgn0039396/FBgn0037976/FBgn0264002/FBgn0031770/FBgn0037408/FBgn0035092/FBgn0035385/FBgn0266429/FBgn0260753/FBgn0045038
## GO:0030055                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 FBgn0034540/FBgn0032683/FBgn0000464/FBgn0010395/FBgn0031850/FBgn0265991/FBgn0033987/FBgn0004397/FBgn0261574/FBgn0004456/FBgn0053993/FBgn0052406
## GO:0005925                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     FBgn0000464/FBgn0010395/FBgn0265991/FBgn0033987/FBgn0004397/FBgn0261574/FBgn0004456/FBgn0053993/FBgn0052406</code></pre>
<p>The results show the analysis identified 859 significantly enriched GO terms, from a ranked list of 15,156 genes, across Biological Process, Molecular Function, and Cellular Component categories.</p>
<p>Not all the genes have a GO-term, this can be because of incomplete annotation, this means that some genes might not be fully annotated in the GO database so their biological functions, processes, or cellular components are not yet known or recorded. It can also be because the GO database does not cover all possible gene functions and processes, especially for less studied organisms or less characterized genes.</p>
<details>
<summary>
An Explanation of the Key Columns
</summary>
<ul>
<li><strong>ONTOLOGY</strong>: The GO category (Biological Process, Molecular Function, or Cellular Component).</li>
<li><strong>ID</strong>: Unique identifier for each GO term.</li>
<li><strong>Description</strong>: The meaning of the GO term and what it represents.</li>
<li><strong>setSize</strong>: Number of genes associated with each GO term.</li>
<li><strong>Enrichment Score (ES)</strong>: Shows if the GO term is overrepresented at the extremes of the gene list. Negative values indicate downregulated genes, while positive values indicate upregulated genes.</li>
<li><strong>Normalized Enrichment Score (NES)</strong>: Adjusted ES to account for gene set size and correlations. Higher absolute NES values mean stronger enrichment.</li>
<li><strong>pvalue, p.adjust, qvalue</strong>: Measures of significance. p-value is the raw score, p.adjust is adjusted for multiple tests, and qvalue shows the false discovery rate.</li>
<li><strong>rank</strong>: Position of the important genes within the list.</li>
<li><strong>leading_edge</strong>: Genes contributing most to the enrichment signal.</li>
<li><strong>core_enrichment</strong>: Key genes driving the enrichment within the leading edge.</li>
</ul>
</details>
<p><strong>Dotplot</strong></p>
<p>We can visualise the enriched GO terms with the dotplot() function within Clusterprofiler. Facet grid is used to split the two plots into two seperate panels. This makes it easier to compare patterns.</p>
<p>I can see the activated and suppressed biological processes that the genes are involved in.
You can also see what processes are more present in the gene set, for example a lot of genes are involved in <strong>G protein-coupled receptor signaling pathway</strong>, a total of 227 genes (see the gene_set_enrichment table).</p>
<p><strong>Low p-adjust value</strong>: Indicates strong statistical significance, suggesting the GO term is likely to be genuinely enriched and relevant to the gene set.
It implies that the observed enrichment is unlikely to be due to random chance.</p>
<p><strong>High p-adjust value</strong>: Indicates weak or no statistical significance, suggesting the enrichment of the GO term may be due to random variation rather than a true biological effect.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb7-1" tabindex="-1"></a><span class="fu">require</span>(DOSE)</span>
<span id="cb7-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb7-2" tabindex="-1"></a><span class="fu">dotplot</span>(gene_set_enrichment, <span class="at">showCategory=</span> <span class="dv">4</span>, <span class="at">split=</span><span class="st">&quot;.sign&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign)</span></code></pre></div>
<p><img src="Images/nonono-1.png" width="672" /></p>
<p><strong>Ridgeplot</strong></p>
<p>We can also visualize the GO-terms with a visualisation specifically for GSEA only: the Ridgeplot.
Basically it groups gene sets and displays density plots for each, showing the distribution of fold change values.
This helps quickly see which pathways are upregulated or downregulated and provides an overview of gene involvement in various biological processes.</p>
<p>Here we also see that the GO term G protein-coupled receptor signaling pathway is upregulated.
On the y-as the GO-terms are visible, on the x-as the padjust values.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb8-1" tabindex="-1"></a><span class="co"># Performing ridgeplot </span></span>
<span id="cb8-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb8-2" tabindex="-1"></a><span class="fu">ridgeplot</span>(gene_set_enrichment, <span class="at">showCategory =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb8-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb8-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Enrichment distribution in the GO-terms for Drosophila data&quot;</span>)</span></code></pre></div>
<p><img src="Images/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="kegg-gene-set-enrichment-analysis-what-pathways-are-enriched" class="section level3 hasAnchor" number="2.7.4">
<h3><span class="header-section-number">2.7.4</span> KEGG Gene Set Enrichment Analysis: What pathways are enriched?<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#kegg-gene-set-enrichment-analysis-what-pathways-are-enriched" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To find KEGG pathways related to our genes using <code>gseKEGG()</code>, we first need to change how our gene identifiers are labeled.
We do this with the <code>bitr</code> function in clusterProfiler.
After this, we use our original gene list to start the analysis.</p>
</div>
<div id="prepare-input" class="section level3 hasAnchor" number="2.7.5">
<h3><span class="header-section-number">2.7.5</span> Prepare Input<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#prepare-input" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Pseudocoding:</p>
<p>- Convert ENSEMBL IDs to ENTREZ IDs.</p>
<p>- Remove duplicate ENSEMBL IDs.</p>
<p>- Select successfully mapped genes.</p>
<p>- Add ENTREZ IDs to the selected genes.</p>
<p>- Create a gene universe vector based on log2 fold change values.</p>
<p>- Label the vector with ENTREZ IDs.</p>
<p>- Remove NA values.</p>
<p>- Sort the vector in decreasing order.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-1" tabindex="-1"></a><span class="co"># Convert gene IDs for gseKEGG function</span></span>
<span id="cb9-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-2" tabindex="-1"></a>gene_id_mapping <span class="ot">&lt;-</span> <span class="fu">bitr</span>(<span class="fu">names</span>(input_gene_list), <span class="at">fromType =</span> <span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb =</span> organism)</span>
<span id="cb9-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-3" tabindex="-1"></a></span>
<span id="cb9-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-4" tabindex="-1"></a><span class="co"># Remove duplicate IDs based on the ENSEMBL column</span></span>
<span id="cb9-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-5" tabindex="-1"></a>unique_gene_ids <span class="ot">&lt;-</span> gene_id_mapping <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(ENSEMBL, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-6" tabindex="-1"></a></span>
<span id="cb9-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-7" tabindex="-1"></a><span class="co"># Filter drosphilia_data to include only the genes that were successfully mapped. So filtered_gene_data contains only the rows from drosphilia_data where the &quot;X&quot; column matches the &quot;ENSEMBL&quot; column in unique_gene_ids</span></span>
<span id="cb9-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-8" tabindex="-1"></a>filtered_gene_data <span class="ot">&lt;-</span> drosphilia_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(X <span class="sc">%in%</span> unique_gene_ids<span class="sc">$</span>ENSEMBL)</span>
<span id="cb9-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-10" tabindex="-1"></a><span class="co"># Add a new column to filtered_gene_data with the corresponding ENTREZ IDs</span></span>
<span id="cb9-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-11" tabindex="-1"></a>filtered_gene_data<span class="sc">$</span>ENTREZID <span class="ot">&lt;-</span> unique_gene_ids<span class="sc">$</span>ENTREZID</span>
<span id="cb9-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-12" tabindex="-1"></a></span>
<span id="cb9-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-13" tabindex="-1"></a><span class="co"># Create a vector of log2FoldChange values for the gene universe</span></span>
<span id="cb9-14"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-14" tabindex="-1"></a>kegg_gene_list <span class="ot">&lt;-</span> filtered_gene_data<span class="sc">$</span>log2FoldChange</span>
<span id="cb9-15"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-15" tabindex="-1"></a></span>
<span id="cb9-16"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-16" tabindex="-1"></a><span class="co"># Name the vector with ENTREZ IDs</span></span>
<span id="cb9-17"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-17" tabindex="-1"></a><span class="fu">names</span>(kegg_gene_list) <span class="ot">&lt;-</span> filtered_gene_data<span class="sc">$</span>ENTREZID</span>
<span id="cb9-18"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-18" tabindex="-1"></a></span>
<span id="cb9-19"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-19" tabindex="-1"></a><span class="co"># Remove any NA values from the vector</span></span>
<span id="cb9-20"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-20" tabindex="-1"></a>kegg_gene_list <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(kegg_gene_list)</span>
<span id="cb9-21"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-21" tabindex="-1"></a></span>
<span id="cb9-22"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-22" tabindex="-1"></a><span class="co"># Sort the list in decreasing order (required for clusterProfiler)</span></span>
<span id="cb9-23"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb9-23" tabindex="-1"></a>kegg_gene_list <span class="ot">&lt;-</span> <span class="fu">sort</span>(kegg_gene_list, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Values: <code>kegg_gene_list</code> contains the log2 fold change values from the gene expression analysis.
These values indicate how much the expression of each gene has changed.</p>
<p>Names: The <code>kegg_gene_list</code> vector is labeled with the corresponding ENTREZ IDs of the genes.
This means that each value in the vector is linked to a specific gene, identified by its ENTREZ ID.</p>
</div>
<div id="create-gsekegg-object-what-pathways-are-enriched" class="section level3 hasAnchor" number="2.7.6">
<h3><span class="header-section-number">2.7.6</span> Create gseKEGG Object: What pathways are enriched?<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#create-gsekegg-object-what-pathways-are-enriched" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The gseKEGG function evaluates the association between the kegg gene list and KEGG pathways and does so by making an object, the gseKEGG object.
To create the gseKEGG object for our analysis, we need to set several parameters.</p>
<details>
<summary>
Parameters
</summary>
<ul>
<li>
<strong>kegg_organism</strong>: The KEGG organism code, a three-letter code identifying the organism. This is used later for generating pathview plots.
</li>
<li>
<strong>nPerm</strong>: Number of permutations in the analysis. More permutations can improve accuracy but take longer to compute.
</li>
<li>
<strong>minGSSize</strong>: Minimum number of genes in each gene set for the analysis.
</li>
<li>
<strong>maxGSSize</strong>: Maximum number of genes allowed in the gene sets for testing.
</li>
<li>
<strong>pvalueCutoff</strong>: Cutoff value for determining statistical significance.
</li>
<li>
<strong>pAdjustMethod</strong>: Method for adjusting p-values. Options include “holm”, “hochberg”, “hommel”, “bonferroni”, “BH”, “BY”, “fdr”, and “none”.
</li>
<li>
<strong>keyType</strong>: Type of gene identifiers used. Options are ‘kegg’, ‘ncbi-geneid’, ‘ncbi-proteinid’, or ‘uniprot’.
</li>
</ul>
</details>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-1" tabindex="-1"></a>kegg_organism <span class="ot">=</span> <span class="st">&quot;dme&quot;</span></span>
<span id="cb10-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-2" tabindex="-1"></a>kegg_object <span class="ot">&lt;-</span> <span class="fu">gseKEGG</span>(<span class="at">geneList     =</span> kegg_gene_list,</span>
<span id="cb10-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-3" tabindex="-1"></a>               <span class="at">organism     =</span> kegg_organism,</span>
<span id="cb10-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-4" tabindex="-1"></a>               <span class="at">nPerm        =</span> <span class="dv">10000</span>,</span>
<span id="cb10-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-5" tabindex="-1"></a>               <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb10-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-6" tabindex="-1"></a>               <span class="at">maxGSSize    =</span> <span class="dv">800</span>,</span>
<span id="cb10-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-7" tabindex="-1"></a>               <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb10-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-8" tabindex="-1"></a>               <span class="at">pAdjustMethod =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb10-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-9" tabindex="-1"></a>               <span class="at">keyType       =</span> <span class="st">&quot;ncbi-geneid&quot;</span>)</span>
<span id="cb10-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-10" tabindex="-1"></a>            </span>
<span id="cb10-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-11" tabindex="-1"></a></span>
<span id="cb10-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-12" tabindex="-1"></a><span class="co"># Displays the top 10 results from the gseKEGG object, showing the most significantly enriched KEGG pathways</span></span>
<span id="cb10-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb10-13" tabindex="-1"></a><span class="fu">head</span>(kegg_object, <span class="dv">10</span>)</span></code></pre></div>
<pre><code>##                ID
## dme04080 dme04080
## dme00053 dme00053
## dme00511 dme00511
## dme00330 dme00330
## dme04310 dme04310
## dme00071 dme00071
## dme00830 dme00830
## dme04130 dme04130
## dme00380 dme00380
## dme03410 dme03410
##                                                                              Description
## dme04080   Neuroactive ligand-receptor interaction - Drosophila melanogaster (fruit fly)
## dme00053         Ascorbate and aldarate metabolism - Drosophila melanogaster (fruit fly)
## dme00511                  Other glycan degradation - Drosophila melanogaster (fruit fly)
## dme00330           Arginine and proline metabolism - Drosophila melanogaster (fruit fly)
## dme04310                     Wnt signaling pathway - Drosophila melanogaster (fruit fly)
## dme00071                    Fatty acid degradation - Drosophila melanogaster (fruit fly)
## dme00830                        Retinol metabolism - Drosophila melanogaster (fruit fly)
## dme04130 SNARE interactions in vesicular transport - Drosophila melanogaster (fruit fly)
## dme00380                     Tryptophan metabolism - Drosophila melanogaster (fruit fly)
## dme03410                      Base excision repair - Drosophila melanogaster (fruit fly)
##          setSize enrichmentScore       NES       pvalue     p.adjust     qvalue
## dme04080      60      -0.6214491 -1.963058 0.0004930156 0.0004930156 0.06435151
## dme00053      32      -0.6646835 -1.853078 0.0022783035 0.0022783035 0.14868928
## dme00511      21      -0.6797063 -1.742066 0.0060305073 0.0060305073 0.18630188
## dme00330      47      -0.5419275 -1.635036 0.0085484412 0.0085484412 0.18630188
## dme04310      92       0.4194541  1.533471 0.0098851189 0.0098851189 0.18630188
## dme00071      32      -0.5858576 -1.633319 0.0119172801 0.0119172801 0.18630188
## dme00830      31      -0.5867306 -1.623965 0.0129733520 0.0129733520 0.18630188
## dme04130      20       0.6327935  1.683913 0.0132966529 0.0132966529 0.18630188
## dme00380      18      -0.6744193 -1.677427 0.0134275618 0.0134275618 0.18630188
## dme03410      26      -0.6112283 -1.634350 0.0142731278 0.0142731278 0.18630188
##          rank                   leading_edge
## dme04080 1364  tags=37%, list=9%, signal=33%
## dme00053   60  tags=16%, list=0%, signal=16%
## dme00511 1575 tags=57%, list=11%, signal=51%
## dme00330 2290 tags=26%, list=16%, signal=22%
## dme04310 2391 tags=27%, list=16%, signal=23%
## dme00071 2984 tags=44%, list=20%, signal=35%
## dme00830 2024 tags=26%, list=14%, signal=22%
## dme04130 2941 tags=50%, list=20%, signal=40%
## dme00380  290  tags=22%, list=2%, signal=22%
## dme03410 1580 tags=35%, list=11%, signal=31%
##                                                                                                                                                core_enrichment
## dme04080                 41017/39409/37892/40955/37191/33360/42530/34878/33248/43669/43551/36601/41639/37004/43484/41726/2768688/36368/44484/43838/38742/36475
## dme00053                                                                                                                         53584/53507/35138/34256/35139
## dme00511                                                                               34204/37211/34203/34437/37490/32726/41913/38528/34206/35989/41457/42178
## dme00330                                                                             38986/326112/40026/46717/43620/34495/41167/40028/50001/43625/326188/34256
## dme04310 39596/34011/45307/45343/32188/37713/31151/48311/42692/31310/34367/31597/40090/44317/34819/33204/43769/43319/31659/39884/34010/39751/32132/38890/31014
## dme00071                                                                 37446/34276/34315/34313/3771877/41311/37445/41480/40059/42364/31695/37217/35213/34256
## dme00830                                                                                                     53513/3771877/41311/38598/53584/53507/35138/35139
## dme04130                                                                                          36080/40094/42215/36015/248102/38614/40373/38541/40724/39051
## dme00380                                                                                                                               31695/43689/35190/34256
## dme03410                                                                                                 34423/31451/42758/43796/34792/35365/39746/40994/33500</code></pre>
</div>
<div id="visualising-the-most-enriched-pathways" class="section level3 hasAnchor" number="2.7.7">
<h3><span class="header-section-number">2.7.7</span> Visualising the most enriched pathways<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#visualising-the-most-enriched-pathways" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After creating the gseKEGG object, we proceed to visualize the enriched pathways using a dotplot.
Each dot represents a pathway, and the position and size of the dots indicate the significance and number of genes involved in each pathway, respectively.
The dotplot allows for easy comparison and interpretation of pathway enrichment results.</p>
<p><strong>Dotplot</strong></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb12-1" tabindex="-1"></a><span class="fu">dotplot</span>(kegg_object, <span class="at">showCategory =</span> <span class="dv">3</span>, <span class="at">title =</span> <span class="st">&quot;Enriched Pathways in the Drosophila melanogaster&quot;</span> , <span class="at">split=</span><span class="st">&quot;.sign&quot;</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign)</span></code></pre></div>
<p><img src="Images/plotje-1.png" width="672" /></p>
<p><strong>Cnet plot</strong></p>
<p>A <code>cnetplot</code> is a type of chart that shows how different genes are connected to important pathways in your analysis.
It uses circles for genes and pathways, with lines between them to show which genes are involved in which pathways.
If a gene has lots of lines, it means it plays a big role in many pathways.
This helps you see which genes are most important and how different pathways are related to each other.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb13-1" tabindex="-1"></a><span class="co"># Generate cnetplot</span></span>
<span id="cb13-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb13-2" tabindex="-1"></a><span class="fu">cnetplot</span>(kegg_object, <span class="at">showCategory =</span> <span class="dv">3</span>, <span class="at">foldChange =</span> kegg_gene_list)</span></code></pre></div>
<p><img src="Images/unnamed-chunk-8-1.png" width="672" /></p>
</div>
<div id="pathview" class="section level3 hasAnchor" number="2.7.8">
<h3><span class="header-section-number">2.7.8</span> Pathview<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#pathview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Pathview() creates visual maps of enriched KEGG pathways, making it easier to understand gene expression data. You can choose any of the enriched pathways, I chose Snare interactions in vesicular transport as this was one of the enriched activated pathways in my dataset.</p>
<details>
<summary>
Parameters
</summary>
<ul>
<li><p>gene.data: This parameter corresponds to the kegg_gene_list created earlier in our analysis.
It contains the gene expression data necessary for pathway visualization.</p></li>
<li><p>pathway.id: The input of the pathway ID of interest.
Information on enriched pathways, including their IDs, can be found in the output table of the gseKEGG function.</p></li>
<li><p>species: This parameter should be the same as the organism specified in the gseKEGG function, which we defined earlier as kegg_organism.
It denotes the KEGG organism code for the species under investigation.</p>
<details></li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-1" tabindex="-1"></a><span class="fu">library</span>(pathview)</span>
<span id="cb14-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-3" tabindex="-1"></a><span class="co"># Produce the native KEGG plot</span></span>
<span id="cb14-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-4" tabindex="-1"></a>dme <span class="ot">&lt;-</span> <span class="fu">pathview</span>(<span class="at">gene.data=</span></span>
<span id="cb14-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-5" tabindex="-1"></a>                  kegg_gene_list, </span>
<span id="cb14-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-6" tabindex="-1"></a>                <span class="at">pathway.id=</span><span class="st">&quot;dme04130&quot;</span>, </span>
<span id="cb14-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb14-7" tabindex="-1"></a>                <span class="at">species =</span> kegg_organism)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="Images/dme04130.pathview.png" alt="Pathway snare interactions in vescular transport in Drosophila melanogaster" width="70%" />
<p class="caption">
Figure 2.3: Pathway snare interactions in vescular transport in Drosophila melanogaster
</p>
</div>
<hr />
</div>
</div>
<div id="ora-with-the-airway-dataset" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> ORA with the airway dataset<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#ora-with-the-airway-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, I will try to do the pathway analysis on the airway data with the over-representation analysis.
I’m executing the same steps I did with the Drosophila melanogaster annotation data.
First I need the results of the differential expression (DE).
There are different ways to get a DESeqDataSet.
Since I am using a SummarizedExperiment, the airway dataset, it can be converted to a DESeq2.
I followed the steps as described <a href="https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#se">here</a>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-1" tabindex="-1"></a><span class="co">#Load the libraries </span></span>
<span id="cb15-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-2" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)  </span>
<span id="cb15-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-3" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb15-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-5" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb15-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-6" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb15-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-7" tabindex="-1"></a><span class="fu">library</span>(airway)</span>
<span id="cb15-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-8" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb15-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-9" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb15-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-10" tabindex="-1"></a><span class="fu">library</span>(ggplotify)</span>
<span id="cb15-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-11" tabindex="-1"></a></span>
<span id="cb15-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-12" tabindex="-1"></a><span class="co"># Loading the built in airway data </span></span>
<span id="cb15-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-13" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;airway&quot;</span>)</span>
<span id="cb15-14"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-14" tabindex="-1"></a></span>
<span id="cb15-15"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-15" tabindex="-1"></a><span class="co"># Creating Deseq2 object </span></span>
<span id="cb15-16"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-16" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(airway, <span class="at">design =</span> <span class="sc">~</span> cell <span class="sc">+</span> dex)</span>
<span id="cb15-17"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-17" tabindex="-1"></a></span>
<span id="cb15-18"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-18" tabindex="-1"></a><span class="co"># Deseq2 analysis and obtaining the results </span></span>
<span id="cb15-19"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-19" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb15-20"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-20" tabindex="-1"></a>results_dds <span class="ot">&lt;-</span> <span class="fu">results</span>(dds) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb15-21"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-21" tabindex="-1"></a></span>
<span id="cb15-22"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-22" tabindex="-1"></a><span class="co"># Naming the Ensemble column by making new column based on the ensemble column that was already present</span></span>
<span id="cb15-23"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-23" tabindex="-1"></a>results_dds<span class="sc">$</span>Ensemble <span class="ot">&lt;-</span> <span class="fu">rownames</span>(results_dds)</span>
<span id="cb15-24"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-24" tabindex="-1"></a></span>
<span id="cb15-25"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-25" tabindex="-1"></a><span class="co"># Checking the first few lines </span></span>
<span id="cb15-26"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb15-26" tabindex="-1"></a><span class="fu">head</span>(results_dds)</span></code></pre></div>
<pre><code>##                    baseMean log2FoldChange     lfcSE       stat       pvalue
## ENSG00000000003 708.6021697     0.38125398 0.1006544  3.7877523 0.0001520163
## ENSG00000000005   0.0000000             NA        NA         NA           NA
## ENSG00000000419 520.2979006    -0.20681260 0.1122186 -1.8429433 0.0653372915
## ENSG00000000457 237.1630368    -0.03792043 0.1434447 -0.2643558 0.7915057416
## ENSG00000000460  57.9326331     0.08816818 0.2871418  0.3070545 0.7588019240
## ENSG00000000938   0.3180984     1.37822703 3.4998728  0.3937935 0.6937335303
##                        padj        Ensemble
## ENSG00000000003 0.001282918 ENSG00000000003
## ENSG00000000005          NA ENSG00000000005
## ENSG00000000419 0.196469846 ENSG00000000419
## ENSG00000000457 0.911419623 ENSG00000000457
## ENSG00000000460 0.895004783 ENSG00000000460
## ENSG00000000938          NA ENSG00000000938</code></pre>
<p>After obtaining the Deseq2 object, we will now create a gene list of up- and downregulated genes.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-1" tabindex="-1"></a><span class="co"># Selecting only significant genes, so padjusted value needs to be &lt;0.05</span></span>
<span id="cb17-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-2" tabindex="-1"></a>significant_DE_genes <span class="ot">&lt;-</span> <span class="fu">filter</span>(results_dds, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb17-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-3" tabindex="-1"></a></span>
<span id="cb17-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-4" tabindex="-1"></a><span class="co"># Selecting the genes that are downregulated (so &lt;-1), selecting genes that are downregulated AND that have a significant expression </span></span>
<span id="cb17-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-5" tabindex="-1"></a>sig_down_genes <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(results_dds, padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb17-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-6" tabindex="-1"></a></span>
<span id="cb17-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-7" tabindex="-1"></a><span class="co"># Selecting genes that are strongly upregulated AND significant expression </span></span>
<span id="cb17-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-8" tabindex="-1"></a>sig_up_genes <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(results_dds, padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb17-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-9" tabindex="-1"></a></span>
<span id="cb17-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-10" tabindex="-1"></a><span class="co"># Rank results for GSEA. This sorts the genelist based on logFoldChange, essential for GSEA.</span></span>
<span id="cb17-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-11" tabindex="-1"></a>GSEA_data <span class="ot">&lt;-</span> results_dds <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(padj)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(log2FoldChange)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb17-12" tabindex="-1"></a>  <span class="fu">pull</span>(log2FoldChange, Ensemble) <span class="co"># output= a vector with genes and their log fold changes</span></span></code></pre></div>
<div id="go-term-enrichment-ontology-analysis-1" class="section level3 hasAnchor" number="2.8.1">
<h3><span class="header-section-number">2.8.1</span> GO term enrichment ontology analysis<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#go-term-enrichment-ontology-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ClusterProfiler offers support for gene ID conversion using the functions bitr() and bitr_kegg().
bitr() uses the OrgDb packages, of which there are 19 databases like the Drosphilia earlier.
Since Clusterprofiler also supports ENSEMBL, I will leave it like this and use this as input for enrichGO() for the GO term ontology analysis. Let’s also say I am not interested in the Molecular Function of the genes (MF) or the Cellular Component, the location of the genes in the cell, but the BP (Biological Process) ontology. This ontology focuses on biological processes, which are the series of events or molecular functions carried out by gene products, like a pathway or a cellular process.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb18-1" tabindex="-1"></a><span class="co"># Obtain keytype info</span></span>
<span id="cb18-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb18-2" tabindex="-1"></a><span class="fu">keytypes</span>(org.Hs.eg.db)</span></code></pre></div>
<pre><code>##  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot;  &quot;ENSEMBLTRANS&quot;
##  [6] &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;     &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;
## [11] &quot;GENETYPE&quot;     &quot;GO&quot;           &quot;GOALL&quot;        &quot;IPI&quot;          &quot;MAP&quot;
## [16] &quot;OMIM&quot;         &quot;ONTOLOGY&quot;     &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;
## [21] &quot;PMID&quot;         &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;
## [26] &quot;UNIPROT&quot;</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-1" tabindex="-1"></a><span class="co"># GO enrichment on the up-regulated genes</span></span>
<span id="cb20-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-2" tabindex="-1"></a>GO_enrich_up <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> sig_up_genes<span class="sc">$</span>Ensemble,</span>
<span id="cb20-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-3" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb20-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-4" tabindex="-1"></a>                <span class="at">keyType =</span> <span class="st">&quot;ENSEMBL&quot;</span>,</span>
<span id="cb20-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-5" tabindex="-1"></a>                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,  <span class="co"># Biological Process ontology</span></span>
<span id="cb20-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-6" tabindex="-1"></a>                <span class="at">minGSSize =</span> <span class="dv">5</span>,  </span>
<span id="cb20-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-7" tabindex="-1"></a>                <span class="at">maxGSSize =</span> <span class="dv">1000</span>) </span>
<span id="cb20-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-8" tabindex="-1"></a></span>
<span id="cb20-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-9" tabindex="-1"></a></span>
<span id="cb20-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-10" tabindex="-1"></a>simple_GO_enrich_up <span class="ot">&lt;-</span>clusterProfiler<span class="sc">::</span><span class="fu">simplify</span>(GO_enrich_up)</span>
<span id="cb20-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb20-11" tabindex="-1"></a>simple_GO_enrich_up</span></code></pre></div>
<pre><code>## #
## # over-representation test
## #
## #...@organism     Homo sapiens
## #...@ontology     BP
## #...@keytype      ENSEMBL
## #...@gene     chr [1:483] &quot;ENSG00000004846&quot; &quot;ENSG00000005471&quot; &quot;ENSG00000006283&quot; ...
## #...pvalues adjusted by &#39;BH&#39; with cutoff &lt;0.05
## #...222 enriched terms found
## &#39;data.frame&#39;:    222 obs. of  9 variables:
##  $ ID         : chr  &quot;GO:0120039&quot; &quot;GO:0048858&quot; &quot;GO:0048812&quot; &quot;GO:0045597&quot; ...
##  $ Description: chr  &quot;plasma membrane bounded cell projection morphogenesis&quot; &quot;cell projection morphogenesis&quot; &quot;neuron projection morphogenesis&quot; &quot;positive regulation of cell differentiation&quot; ...
##  $ GeneRatio  : chr  &quot;41/397&quot; &quot;41/397&quot; &quot;40/397&quot; &quot;50/397&quot; ...
##  $ BgRatio    : chr  &quot;700/21261&quot; &quot;705/21261&quot; &quot;683/21261&quot; &quot;999/21261&quot; ...
##  $ pvalue     : num  1.08e-10 1.34e-10 1.85e-10 2.25e-10 2.33e-09 ...
##  $ p.adjust   : num  2.64e-07 2.64e-07 2.64e-07 2.64e-07 1.27e-06 ...
##  $ qvalue     : num  2.07e-07 2.07e-07 2.07e-07 2.07e-07 9.97e-07 ...
##  $ geneID     : chr  &quot;ENSG00000007237/ENSG00000046889/ENSG00000049759/ENSG00000055163/ENSG00000061337/ENSG00000066468/ENSG00000075213&quot;| __truncated__ &quot;ENSG00000007237/ENSG00000046889/ENSG00000049759/ENSG00000055163/ENSG00000061337/ENSG00000066468/ENSG00000075213&quot;| __truncated__ &quot;ENSG00000007237/ENSG00000046889/ENSG00000049759/ENSG00000055163/ENSG00000061337/ENSG00000066468/ENSG00000075213&quot;| __truncated__ &quot;ENSG00000064309/ENSG00000073756/ENSG00000078725/ENSG00000092969/ENSG00000102468/ENSG00000103257/ENSG00000104951&quot;| __truncated__ ...
##  $ Count      : int  41 41 40 50 32 36 35 22 37 33 ...
## #...Citation
##  T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu.
##  clusterProfiler 4.0: A universal enrichment tool for interpreting omics data.
##  The Innovation. 2021, 2(3):100141</code></pre>
<p>The results show that 483 human genes (Homo sapiens) were examined using an over-representation test focused on the <strong>Biological Process (BP)</strong> ontology. The results identified 222 significantly enriched GO terms, describing processes like “cell projection morphogenesis” and “positive regulation of cell differentiation.” The p-values were adjusted using the Benjamini-Hochberg (BH) method to reduce false positives. These results highlight which biological processes might be involved with the genes studied.</p>
</div>
<div id="visualize-enriched-go-terms" class="section level3 hasAnchor" number="2.8.2">
<h3><span class="header-section-number">2.8.2</span> Visualize enriched GO-terms<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#visualize-enriched-go-terms" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Dotplot</strong></p>
<p>In the dotplot you can see the enriched GO-terms that are associated with the upregulated genes in the dataset. The gene ratio is the ratio that represents the proportion of genes from your analyzed gene list that are associated with a specific Gene Ontology (GO) term or pathway. Calculated as: gene ratio= number of genes in the list associated with a GO-term/total number of genes in the list.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb22-1" tabindex="-1"></a><span class="co"># dotplot for the upregulated genes </span></span>
<span id="cb22-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb22-2" tabindex="-1"></a><span class="fu">dotplot</span>(simple_GO_enrich_up,<span class="at">showCategory=</span><span class="dv">20</span>,<span class="at">font.size=</span><span class="dv">10</span>,<span class="at">label_format=</span><span class="dv">70</span>)<span class="sc">+</span></span>
<span id="cb22-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb22-3" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">7</span>))<span class="sc">+</span> </span>
<span id="cb22-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb22-4" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb22-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb22-5" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;GO Enrichment of up-regulated genes&quot;</span>)</span></code></pre></div>
<p><img src="Images/kanarie-1.png" width="672" /></p>
<p><strong>Barplot</strong></p>
<p>The bar plot below displays the enriched Gene Ontology (GO) terms for the airway data.
GO terms represent biological processes, molecular functions, or cellular components that genes are involved in.
By setting the <code>showCategory</code> parameter, you can specify the number of top terms to display.
In this case, I set <code>showCategory=10</code>, so the plot shows the 10 most significant GO terms and the number of genes from the airway dataset associated with each term (Count).</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb23-1" tabindex="-1"></a><span class="co"># Barplot for GO-terms</span></span>
<span id="cb23-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb23-2" tabindex="-1"></a><span class="fu">barplot</span>(simple_GO_enrich_up, <span class="at">showCategory =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="Images/unnamed-chunk-11-1.png" width="672" /></p>
<p><strong>Go plot</strong></p>
<p>The <code>goplot()</code> function visualizes GO enrichment analysis results, showing connections between genes and GO terms.
It highlights the most significant biological processes, helping to quickly identify key genes and pathways involved.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb24-1" tabindex="-1"></a><span class="co"># Performing goplot </span></span>
<span id="cb24-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb24-2" tabindex="-1"></a><span class="fu">goplot</span>(simple_GO_enrich_up, <span class="at">showCategory =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="Images/unnamed-chunk-12-1.png" width="672" /></p>
<p><strong>Cnetplot</strong></p>
<p>The <code>cnetplot()</code> function helps visualize which genes are involved in significant enriched terms, unlike <code>barplot()</code> and <code>dotplot()</code>, which only show the most significant terms.
<code>cnetplot()</code> shows the complex associations between genes and biological concepts (like GO terms or KEGG pathways) in a network, highlighting how genes may belong to multiple categories.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-1" tabindex="-1"></a><span class="co"># Color the genes by log2 fold changes</span></span>
<span id="cb25-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> sig_up_genes<span class="sc">$</span>log2FoldChange</span>
<span id="cb25-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-3" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">&lt;-</span> sig_up_genes<span class="sc">$</span>Ensemble</span>
<span id="cb25-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-4" tabindex="-1"></a></span>
<span id="cb25-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-5" tabindex="-1"></a><span class="co"># The cnetplot gives the top5 terms by default. If you want to see which genes are involved in GO-terms 1,2 and 10, you can do this by making a subset</span></span>
<span id="cb25-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-6" tabindex="-1"></a>simple_GO_cnet <span class="ot">&lt;-</span> simple_GO_enrich_up</span>
<span id="cb25-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-7" tabindex="-1"></a>simple_GO_cnet<span class="sc">@</span>result<span class="ot">&lt;-</span>simple_GO_enrich_up[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">10</span>),]</span>
<span id="cb25-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-8" tabindex="-1"></a></span>
<span id="cb25-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-9" tabindex="-1"></a><span class="fu">cnetplot</span>(simple_GO_cnet, </span>
<span id="cb25-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-10" tabindex="-1"></a>         <span class="at">foldChange=</span>foldchanges, </span>
<span id="cb25-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-11" tabindex="-1"></a>         <span class="at">shadowtext=</span><span class="st">&#39;gene&#39;</span>,</span>
<span id="cb25-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-12" tabindex="-1"></a>         <span class="at">cex_label_gene=</span><span class="fl">0.25</span>,</span>
<span id="cb25-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-13" tabindex="-1"></a>         <span class="at">cex_label_category=</span><span class="fl">0.5</span>,</span>
<span id="cb25-14"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb25-14" tabindex="-1"></a>         <span class="at">color_category=</span><span class="st">&quot;#60a0b03d&quot;</span>)</span></code></pre></div>
<p><img src="Images/taylor-1.png" width="672" /></p>
</div>
<div id="kegg-pathway-enrichment-analyse" class="section level3 hasAnchor" number="2.8.3">
<h3><span class="header-section-number">2.8.3</span> KEGG pathway enrichment analyse<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#kegg-pathway-enrichment-analyse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-1" tabindex="-1"></a><span class="co"># Convert Ensembl IDs to Entrez IDs</span></span>
<span id="cb26-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-2" tabindex="-1"></a>idds <span class="ot">&lt;-</span> <span class="fu">bitr</span>(sig_up_genes<span class="sc">$</span>Ensemble, <span class="at">fromType =</span> <span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb26-3"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-3" tabindex="-1"></a></span>
<span id="cb26-4"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-4" tabindex="-1"></a><span class="co"># Select only the Entrez IDs column after mapping</span></span>
<span id="cb26-5"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-5" tabindex="-1"></a>entrez_ids <span class="ot">&lt;-</span> idds<span class="sc">$</span>ENTREZID</span>
<span id="cb26-6"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-7" tabindex="-1"></a><span class="co"># Convert Ensemble IDs to Entrez IDs</span></span>
<span id="cb26-8"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-8" tabindex="-1"></a>results_dds_entrez <span class="ot">&lt;-</span> <span class="fu">bitr</span>(<span class="fu">rownames</span>(results_dds), <span class="at">fromType =</span> <span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>, <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb26-9"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-9" tabindex="-1"></a></span>
<span id="cb26-10"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-10" tabindex="-1"></a><span class="co"># Perform KEGG pathway enrichment analysis</span></span>
<span id="cb26-11"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-11" tabindex="-1"></a>kegg_enrichment <span class="ot">&lt;-</span> <span class="fu">enrichKEGG</span>(<span class="at">gene =</span> idds<span class="sc">$</span>ENTREZID,</span>
<span id="cb26-12"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-12" tabindex="-1"></a>                              <span class="at">organism =</span> <span class="st">&#39;hsa&#39;</span>,</span>
<span id="cb26-13"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-13" tabindex="-1"></a>                              <span class="at">universe =</span> results_dds_entrez<span class="sc">$</span>ENTREZID,</span>
<span id="cb26-14"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-14" tabindex="-1"></a>                              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb26-15"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-15" tabindex="-1"></a>                              <span class="at">minGSSize =</span> <span class="dv">1</span>, </span>
<span id="cb26-16"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-16" tabindex="-1"></a>                              <span class="at">maxGSSize =</span> <span class="dv">200000</span>)</span>
<span id="cb26-17"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb26-17" tabindex="-1"></a><span class="fu">head</span>(kegg_enrichment)</span></code></pre></div>
<pre><code>##                                      category
## hsa04750                   Organismal Systems
## hsa04060 Environmental Information Processing
## hsa04713                   Organismal Systems
## hsa04360                   Organismal Systems
## hsa04611                   Organismal Systems
## hsa05200                       Human Diseases
##                                  subcategory       ID
## hsa04750                      Sensory system hsa04750
## hsa04060 Signaling molecules and interaction hsa04060
## hsa04713            Environmental adaptation hsa04713
## hsa04360        Development and regeneration hsa04360
## hsa04611                       Immune system hsa04611
## hsa05200                    Cancer: overview hsa05200
##                                               Description GeneRatio  BgRatio
## hsa04750 Inflammatory mediator regulation of TRP channels    14/213  98/8505
## hsa04060           Cytokine-cytokine receptor interaction    23/213 287/8505
## hsa04713                            Circadian entrainment    12/213  97/8505
## hsa04360                                    Axon guidance    15/213 183/8505
## hsa04611                              Platelet activation    11/213 122/8505
## hsa05200                               Pathways in cancer    27/213 529/8505
##                pvalue     p.adjust       qvalue
## hsa04750 1.269391e-07 3.630458e-05 2.832746e-05
## hsa04060 7.295357e-07 1.043236e-04 8.140083e-05
## hsa04713 4.893891e-06 4.665510e-04 3.640368e-04
## hsa04360 5.212754e-05 3.727119e-03 2.908168e-03
## hsa04611 2.307920e-04 1.320130e-02 1.030061e-02
## hsa05200 3.060702e-04 1.458935e-02 1.138366e-02
##                                                                                                                                        geneID
## hsa04750                                                                     815/623/3356/8989/8605/40/5321/8503/196883/4803/114/107/624/5581
## hsa04060        55540/7042/10673/6387/6354/6355/8743/130399/8744/8200/3976/51554/9518/4803/83729/85480/146433/1230/133396/4982/3592/6356/9966
## hsa04713                                                                          8913/8863/815/9252/196883/8864/5593/2977/114/107/2903/54331
## hsa04360                                                     815/10371/5063/6387/219699/8503/64101/6608/2048/80031/2044/22854/6091/84552/2049
## hsa04611                                                                               8605/1277/5321/8503/196883/5593/2212/2977/114/3673/107
## hsa05200 2263/2255/815/5743/2099/7042/3162/623/9252/7472/6387/7422/8503/1026/6608/196883/10912/183/1030/114/3673/107/624/3592/3725/3714/54331
##          Count
## hsa04750    14
## hsa04060    23
## hsa04713    12
## hsa04360    15
## hsa04611    11
## hsa05200    27</code></pre>
</div>
<div id="visualizing-the-enriched-pathways" class="section level3 hasAnchor" number="2.8.4">
<h3><span class="header-section-number">2.8.4</span> Visualizing the enriched pathways<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#visualizing-the-enriched-pathways" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Dotplot</strong></p>
<p>You can clearly see in the dotplot that “Pathways in Cancer” have the highest Gene ratio and padjust value.
But what does this mean, exactly?</p>
<p>The padjust value in a KEGG pathway analysis represents the adjusted p-value, telling whether the observed enrichment of a specific KEGG pathway is statistically significant after correcting for multiple testing.
A low p-adjust (&lt; 0.05) value means that the enrichment of the KEGG pathway <em>is</em> statistically significant.
So there is high confidence that the association found between the airway dataset and the KEGG pathway is not due to chance.</p>
<p>Whereas a high padjust value (≥ 0.05) indicates that the enrichment of the KEGG pathway is not statistically significant.
In this case, the association between the airway dataset and the KEGG pathway is <em>not</em> considered reliable and could be due to chance.</p>
<p>The “gene ratio” in a KEGG pathway enrichment analysis indicates how many genes from the dataset are involved in a specific KEGG pathway relative to the total number of genes typically associated with that pathway in the reference database (the human KEGG database).
In the table kegg_enrichment, you can see the Gene ratio for the pathway “Pathways in Cancer” is the highest.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb28-1" tabindex="-1"></a><span class="co"># dotplot for enriched pathways in airway dataset </span></span>
<span id="cb28-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb28-2" tabindex="-1"></a><span class="fu">dotplot</span>(kegg_enrichment, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">title=</span> <span class="st">&quot;Enriched Pathways in the Airway Dataset&quot;</span>)</span></code></pre></div>
<p><img src="Images/unnamed-chunk-14-1.png" width="672" /></p>
<p><strong>Barplot</strong></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb29-1" tabindex="-1"></a><span class="co"># Voorbeeld barplot voor pathways </span></span>
<span id="cb29-2"><a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#cb29-2" tabindex="-1"></a><span class="fu">barplot</span>(kegg_enrichment, <span class="at">showCategory =</span> <span class="dv">10</span>)</span></code></pre></div>
<p><img src="Images/unnamed-chunk-15-1.png" width="672" /></p>
<p>The barplot shows pathways linked to the DE genes in the airway study, helping explain their connection to airway inflammation and glucocorticoid treatment.</p>
<ul>
<li><strong>Inflammatory mediator regulation of TRP channels:</strong> Shows how inflammation affects TRP channels, helping understand glucocorticoid action.</li>
<li><strong>Cytokine-cytokine receptor interaction:</strong> Highlights communication between cytokines and receptors, key to managing inflammation with glucocorticoids.</li>
<li><strong>Circadian entrainment:</strong> Suggests the body’s clock may influence airway response to glucocorticoids.</li>
<li><strong>Axon guidance:</strong> Could indicate how airway cells move or react during inflammation, relevant to glucocorticoid treatment.</li>
<li><strong>Platelet activation:</strong> Involves platelets in inflammation, which glucocorticoids can affect.</li>
<li><strong>Pathways in cancer:</strong> Points to similarities between inflamed airway cells and cancer cells in response to glucocorticoids.</li>
</ul>
</div>
<div id="pathview-1" class="section level3 hasAnchor" number="2.8.5">
<h3><span class="header-section-number">2.8.5</span> Pathview<a href="pathway-analysis-using-the-clusterprofiler-package-in-r.html#pathview-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For pathview, I decided to visualize the pathway “Inflammatory mediator regulation of TRP channels”, because of its low padjust value (so higly significant) and the pathway “Cytokine-cytokine receptor interaction”, because it’s highly relevant due to their roles in inflammation and immune response.
This pathway also has a high gene ratio, as it involves 23 out of 213 genes in the dataset.</p>
<p>Even though “Pathways in cancer” has the highest gene ratio (27/213 genes), it also has the highest padjust value, meaning there could be a chance it’s not significant.
Besides, I chose pathways that are directly involved in inflammation, immune response and the airway function, because this helps understanding the disease mechanisms and possible ways to intervene when things go wrong with therapeutics.</p>
<p>I got the pathway ID from the kegg_enrichment table.
hsa means homo sapiens.</p>
<p>First Inflammatory mediator regulation of TRP channels:</p>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-17"></span>
<img src="Images/hsa04750.png" alt="Pathway Inflammatory mediator regulation of TRP channels   " width="70%" />
<p class="caption">
Figure 2.4: Pathway Inflammatory mediator regulation of TRP channels
</p>
</div>
<p>Now Cytokine-cytokine receptor interaction:</p>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="Images/hsa04060.png" alt="Pathway Inflammatory mediator regulation of TRP channels   " width="70%" />
<p class="caption">
Figure 2.5: Pathway Inflammatory mediator regulation of TRP channels
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="future-outlook.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="guerrilla-analytics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
